lib/Interpreter/CMakeLists.txt

```diff
@@ -134,9 +113,37 @@ else()
     ${link_libs}
   )
 
-  target_compile_definitions(clangCppInterOp PUBLIC "_CINDEX_LIB_") # workaround for the use of `CINDEX_LINKAGE`
+
+if(EMSCRIPTEN)
+  # FIXME: When dynamically linking the Emscripten shared library to the 
+  # unit tests main_module you get errors due to undefined symbols. The reading of the file
+  # below into a SYMBOLS_LIST variable is a temporary workaround that exports the undefined
+  # symbols from the shared library, until it can be determined why they are not being exported already.
+  file(READ "${CMAKE_SOURCE_DIR}/lib/Interpreter/exports.ld" SYMBOLS_LIST)
+  
+  # Replace newlines with spaces
+  string(REPLACE "\n" " " SYMBOLS_LIST "${SYMBOLS_LIST}")
+  
+  #FIXME: Setting no_soname=1 is needed until https://github.com/emscripten-core/emscripten/blob/ac676d5e437525d15df5fd46bc2c208ec6d376a3/cmake/Modules/Platform/Emscripten.cmake#L36
+  # is patched out of emsdk, as --soname is not recognised by emscripten. A PR to do this has been done here https://github.com/emscripten-core/emscripten/pull/23453 
+  #FIXME: A patch is needed to llvm to remove -Wl,-z,defs since it is now recognised on emscripten. What needs to be removed is here
+  # https://github.com/llvm/llvm-project/blob/128e2e446e90c3b1827cfc7d4d19e3c0976beff3/llvm/cmake/modules/HandleLLVMOptions.cmake#L318 . The PR to do try to do this is here 
+  # https://github.com/llvm/llvm-project/pull/123396
+  set_target_properties(clangCppInterOp PROPERTIES 
+    NO_SONAME 1
+    LINK_FLAGS "-s WASM_BIGINT  -s SIDE_MODULE=1 ${SYMBOLS_LIST}"
```

Comment:
Hi, 

I see set_target_properties being put to use in this PR.

If possible, can we stick to 
i) target_link_libraries when linking with libraries
ii) target_link_options when setting link options.

Those are the 2 most fundamental functions cmake has to offer when it comes to linking. So we should put them to use wherever possible.